src/ipc/ipc3/helper.c

```diff
@@ -611,16 +595,22 @@ int ipc_comp_connect(struct ipc *ipc, ipc_pipe_comp_connect *_connect)
 	/* check whether the components already exist */
 	icd_source = ipc_get_comp_by_id(ipc, connect->source_id);
 	if (!icd_source) {
-		tr_err(&ipc_tr, "ipc_comp_connect(): source component does not exist, source_id = %u sink_id = %u",
-		       connect->source_id, connect->sink_id);
-		return -EINVAL;
+		icd_source = ipc_get_buffer_by_id(ipc, connect->source_id);
+		if (!icd_source) {
+			tr_err(&ipc_tr, "ipc_comp_connect(): source component does not exist, source_id = %u sink_id = %u",
+			       connect->source_id, connect->sink_id);
+			return -EINVAL;
+		}
```

Comment:
this doesn't make IPC3 code look prettier or run faster. Can we do this: add
```
#define COMP_TYPE_ANY 0
```
and then
```
struct ipc_comp_dev *ipc_get_comp_dev(struct ipc *ipc, uint16_t type, uint32_t id)
{
	struct ipc_comp_dev *icd;
	struct list_item *clist;

	list_for_item(clist, &ipc->comp_list) {
		icd = container_of(clist, struct ipc_comp_dev, list);
		if (icd->id == id && (type == icd->type || type == COMP_TYPE_ANY))
			return icd;
	}

	return NULL;
}
```
and use `ANY` with IPC3 since IDs are unique there anyway?